### ISSUE TITLE
Date of birth input should use input view not calendar view

### ISSUE THREAD

**Issue by MikeMcC399** [corona-warn-app/cwa-app-android#3968]
## Avoid duplicates
* [X] Bug is not mentioned in the [FAQ](https://www.coronawarn.app/en/faq/)
* [X] Bug is specific for Android only, for general issues / questions that apply to iOS and Android please raise them in the [documentation repository](https://github.com/corona-warn-app/cwa-documentation)
* [X] Bug is not already reported in another issue

## Technical details

- Device name: Google Pixel 3a emulator
- Android version: 11
- App version: ~~2.7.1~~ 2.9.0

## Describe the bug

The date picker for date of birth in the Rapid Test Profile and in COVID Test Certificate opens by default in calendar view. The recommended view for dates not near to the current date, like dates of birth, is input view.

The title of the date picker is not set. The default "SELECT DATE" is shown.

## Steps to reproduce the issue

1. Set locale to English (United Kingdom)
2. Tap "Manage Your Tests"
3. Tap "Create Rapid Test Profile" (tap "CONTINUE" if prompted)
4. Tap "Date of Birth"
5. Note that the date picker appears in calendar view, with title "SELECT DATE" and today's date selected
6. Tap "CANCEL", tap back arrow
7. Tap "Scan QR Code", tap "ACCEPT"
8. Scan QR Code e.g. PCR Test QR Code
9. Tap "Date of Birth"
10. Note that the date picker appears in calendar view, with title "SELECT DATE"

![calendar view](https://user-images.githubusercontent.com/66998419/130462081-95e17e01-aa7e-4512-a52c-a303cbd18806.png)

## Expected behavior

Follow the advice from https://material.io/components/date-pickers#mobile-pickers and use the mobile date input picker, not the calendar view.

Add the title "Date of Birth" to the data picker, so the user is reminded what data they are being asked to enter.

## Possible Fix

After https://github.com/corona-warn-app/cwa-app-android/blob/97fbc95b71db96ed637af3eaa35118eb705a1611/Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/coronatest/rat/profile/create/RATProfileCreateFragment.kt#L130-L132

add

```
.setInputMode(MaterialDatePicker.INPUT_MODE_TEXT)
.setTitleText(getResources().getString(R.string.rat_profile_create_birth_date_hint))
```

After https://github.com/corona-warn-app/cwa-app-android/blob/97fbc95b71db96ed637af3eaa35118eb705a1611/Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/submission/covidcertificate/RequestCovidCertificateFragment.kt#L133-L135

add

```
.setInputMode(MaterialDatePicker.INPUT_MODE_TEXT)
.setTitleText(getResources().getString(R.string.request_green_certificate_birthdate_hint))
```

![input view](https://user-images.githubusercontent.com/66998419/130462145-22d064fd-c5a9-4cbe-bca7-426dbf058b27.png)

## Workaround

Select input view from calendar view by tapping the pencil icon.

![image](https://user-images.githubusercontent.com/66998419/130471357-c17c825d-d09f-45f6-9785-1d6311ab8599.png)

## Additional context

https://material.io/components/date-pickers#mobile-pickers says

### Don't
Don’t use a calendar picker to prompt for dates in the distant past or future, **such as a date of birth**. In these cases, use a mobile input picker or a text field instead.

### Do
For dates that don’t require a calendar view, make the mobile date input picker as the default prompt.

---
Internal Tracking ID: [EXPOSUREAPP-9128](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-9128)

**Comment by dsarkar** [comment]
@MikeMcC399 Thanks. Internal Tracking ID: [EXPOSUREAPP-9128](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-9128)

I assume that you have tested this approach. Do you want to prove a PR?

**Comment by MikeMcC399** [comment]
@dsarkar 
Yes, I did already try it out and I can provide a PR if dev would like this submission.

It is an enhancement, not a bug correction. It is just more inconvenient to enter a date of birth from the calendar view.

Let me know if the PR is wanted. If they say no, then that is OK with me.

**Comment by dsarkar** [comment]
@MikeMcC399 ok, thanks for the offer. Stand by, I check with them.

**Comment by dsarkar** [comment]
@MikeMcC399 devs need to check with UX team first. So, currently waiting for their feedback.

**Comment by MikeMcC399** [comment]
PR #3993 
"Change date of birth entry from calendar to input view (EXPOSUREAPP-9128) (COMMUNITY)"
is merged into [release/2.10.x](https://github.com/corona-warn-app/cwa-app-android/tree/release/2.10.x)

**Comment by MikeMcC399** [comment]
Verified on [tag/v2.10.1](https://github.com/corona-warn-app/cwa-app-android/releases/tag/v2.10.1).

Closing as fixed.
